views/swap/manage-slippage/manage-slippage-form/index.tsx

```diff
@@ -0,0 +1,169 @@
+
+  const setSpeed = (value: 'normal' | 'fast' | 'instant') =>
+    formTmpSettings.setValue('speed', value);
+
+  const onConfirm = () => {
+    formSettings.setValue('speed', formTmpSettings.getValues('speed'));
+    formSettings.setValue('slippage', formTmpSettings.getValues('slippage'));
+    formSettings.setValue('deadline', formTmpSettings.getValues('deadline'));
+    handleManageView();
+  };
+
+  return (
+    <Box py="1rem" px="1rem" display="flex" flexDirection="column" gap="1.5rem">
+      <Box>
+        <Typography variant="body" size="small" mb="0.5rem">
+          Slippage Tolerance
+        </Typography>
+        <Box
+          gap="xs"
+          flexDirection="column"
+          gridTemplateColumns="4fr repeat(3, auto)"
+          display={['flex', 'flex', 'flex', 'grid']}
+        >
+          <TextField
+            fontSize="1rem"
+            placeholder="0.1"
+            fontFamily="Satoshi"
+            {...formTmpSettings.register('slippage', {
+              onChange: (v: ChangeEvent<HTMLInputElement>) => {
+                formTmpSettings.setValue?.(
+                  'slippage',
+                  parseInputEventToNumberString(v)
+                );
+              },
+            })}
+            fieldProps={{ borderRadius: 'full', width: '100%' }}
+          />
+          <Button
+            variant="outline"
+            textAlign="center"
+            onClick={() => setTolerance('0.1')}
+          >
+            <Typography variant="label" size="large" width="100%">
+              0.1 %
+            </Typography>
+          </Button>
+          <Button
+            variant="outline"
+            textAlign="center"
+            onClick={() => setTolerance('0.5')}
+          >
+            <Typography variant="label" size="large" width="100%">
+              0.5 %
+            </Typography>
+          </Button>
+          <Button
+            variant="outline"
+            textAlign="center"
+            onClick={() => setTolerance('1')}
+          >
+            <Typography variant="label" size="large" width="100%">
+              1 %
+            </Typography>
+          </Button>
+        </Box>
+      </Box>
+      <Box>
+        <Typography variant="body" size="small" mb="0.5rem">
+          Transaction Speed
+        </Typography>
+        <Box
+          gap="xs"
+          flexDirection="column"
+          gridTemplateColumns="repeat(3, 1fr)"
+          display={['flex', 'flex', 'flex', 'grid']}
+        >
+          <Button
+            variant="outline"
+            textAlign="center"
+            onClick={() => setSpeed('normal')}
+          >
+            <Typography variant="label" size="large" width="100%">
+              Normal
+            </Typography>
+          </Button>
+          <Button
+            variant="outline"
+            textAlign="center"
+            onClick={() => setSpeed('fast')}
+          >
+            <Typography variant="label" size="large" width="100%">
```

Comment:
Why are u putting typography inside of button?

Reply:
to use the variant about Typography